### PR TITLE
#4 Prepare service skeleton

### PR DESCRIPTION
**Pull request by i3odja**
https://barryvanveen.nl/blog/52-solving-inconsistent-line-endings-within-a-git-repository

### CODE REVIEW

**Comment by VasylKosiv** [comment]
I think should be one more package for this issue: Logger compatible with Graylog
https://github.com/lvl484/user-manager/projects/1#card-34310615

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
```
Comment (radikh) [inline .editorconfig]: For go files indent is always tab.

File: swagger-api.yaml [hunk 2]
```diff
@@ -1,233 +1,233 @@
...
-          type: string
-        phone:
-          type: string
-    Error:
-      properties:
-        code:
-          type: string
-        message:
-          type: string
+openapi: "3.0.0"
```
Comment (radikh) [inline swagger-api.yaml]: What are changes in this file?

File: model/user.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package model
+
+import (
+	"time"
+)
+
+type User struct {
+	ID        string    `json:"id"`         // Unique identifier
```
Comment (radikh) [inline model/user.go]: Comments for structure fields should be above the fields not on the same line.

File: model/user.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package model
+
+import (
+	"time"
+)
+
+type User struct {
```
Comment (radikh) [inline model/user.go]: Exported structure has to have a comment.

File: model/user.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	ID        string    `json:"id"`         // Unique identifier
+	Username  string    `json:"user_name"`  // Unique login
+	Password  string    `json:"password"`   // Strong enough password
+	Email     string    `json:"email"`      // Valid email
+	FirstName string    `json:"first_name"` // Obviously first name
+	LastName  string    `json:"last_name"`  // Obviously last name
+	Phone     int       `json:"phone"`      // Valid phone
+	CreatedAt time.Time `json:"created_at"` // Time when user was created
+	UpdatedAt time.Time `json:"updated_at"` // Time of last changes made
```
Comment (radikh) [inline model/user.go]: If you want to do JSON transformation with this model, I would propose to use *time.Time here.

File: .editorconfig (hunk 1 above)
Comment (i3odja) [reply .editorconfig]: I use 4 spaces in this file. Do I need to change on tab?

File: swagger-api.yaml (hunk 2 above)
Comment (i3odja) [reply swagger-api.yaml]: Nothing. It is connected with type of line ending ("lf" or "crlf"). I fixed it with file .gitattributes

File: model/user.go (hunk 5 above)
Comment (i3odja) [reply model/user.go]: I fixed it

File: model/user.go (hunk 4 above)
Comment (i3odja) [reply model/user.go]: Added comment for structure User

File: model/user.go (hunk 3 above)
Comment (i3odja) [reply model/user.go]: I fixed it

File: .editorconfig (hunk 1 above)
Comment (radikh) [reply .editorconfig]: If my understanding is right - yes and no.
Why yes: `[*]` applies rules to all files in the repository, it means that all files will be formatted using 4 spaces indentation, correct me if I'm wrong. For go files spaces are not flavored indentation. 
Why no: Some files like python scripts may require spaces as indentation, so tabs in this rule will not fit them.
Possible solution: define the rule for any type of file explicitly.

File: swagger-api.yaml (hunk 2 above)
Comment (radikh) [reply swagger-api.yaml]: So what's changed in the file, why git shows diffs?

File: .editorconfig (hunk 1 above)
Comment (i3odja) [reply .editorconfig]: I did not understand which values are correct. That is why I deleted .editorconfig file

File: swagger-api.yaml (hunk 2 above)
Comment (i3odja) [reply swagger-api.yaml]: Rollbacked changes in swagger-api.yaml (my IDE changed line endings CRLF).
Fixed.
Thanks

**Comment by i3odja** [comment]
> Comments for packages and other stuff are not really descriptive. Try to read it and understand what does each package do, then, maybe you can make them better then.

Fixed. Please check

File: swagger-api.yaml (hunk 2 above)
Comment (radikh) [reply swagger-api.yaml]: I'm still wondering, what was changed? 🤔

Comment (i3odja) [reply swagger-api.yaml]: To my mind this file was changed because I used Goland in Windows OS. I did not change swagger-api.yaml.

**Comment by radikh** [comment]
Good.

**Pull request merged**